### ISSUE TITLE
Jenkins Changelog Plugin: getHighestSemanticVersion() works only for master branch

### ISSUE THREAD

**Issue by sprnza** [tomasbjerre/git-changelog-lib#127]
Hi! Thanks for the library and the plugin.
I try to get the highest Semantic Version in a custom branch but it seems like it's not possible to specify a branch to gather tags from and it always uses `master` branch. If there is no a master branch it fails with an error.
```
se.bjurr.gitchangelog.api.exceptions.GitChangelogRepositoryException: master not found in:
```
Would be great to have ability to use it like this
```
getHighestSemanticVersion to: [type: 'REF', value: 'HEAD']
```

In the command line there is an option for it and it works as expected:
```
npx git-changelog-command-line -phv -tc HEAD
0.0.2
```

**Comment by sprnza** [comment]
Since getNextSemanticVersion does support specifying `to` this
```
def highestVersion = getHighestSemanticVersion()
def nextVersion = getNextSemanticVersion to: [type: 'REF', value: 'HEAD']
println(highestVersion.toString())
println(nextVersion.toString())
```
yelds to
```
[Pipeline] getHighestSemanticVersion 
[Pipeline] getNextSemanticVersion
[Pipeline] echo
0.0.1
[Pipeline] echo
0.0.5
```

**Comment by tomasbjerre** [comment]
I agree! I changed it so that it will look for `head` and if that does not exist look for `master`. Released the plugin as `3.23`.

**Comment by sprnza** [comment]
Thank you! Works perfectly!
